src/DynamoCore/Models/HomeWorkspaceModel.cs

```diff
@@ -174,9 +167,7 @@ public void ResetEngine(EngineController controller, bool markNodesAsDirty = fal
             
             if (markNodesAsDirty)
             {
-                foreach (var node in Nodes)
-                    node.ForceReExecuteOfNode = true;
-                OnAstUpdated();
+                MarkAllNodesDirty();
```

Comment:
This has the same issue as the `MarkAllNodesDirty` above. Suggest removing it and use only the method call `NodeModel.MarkAsModified` in a `foreach`.